tokio-epoll-uring/src/metrics.rs

```diff
@@ -1,29 +1,68 @@
-use std::sync::atomic::{AtomicU64, Ordering};
+use std::{
+    collections::HashMap,
+    sync::{
+        atomic::{AtomicU64, Ordering},
+        RwLock,
+    },
+};
+
+use once_cell::sync::Lazy;
 
 pub struct Metrics {
     pub systems_created: u64,
     pub systems_destroyed: u64,
+    pub slots_waiters_queue_depth: Vec<u64>,
 }
 
 pub(crate) struct MetricsStorage {
     pub(crate) systems_created: AtomicU64,
     pub(crate) systems_destroyed: AtomicU64,
+    /// Waiters queue depth for active system, indexed by system id.
+    pub(crate) slots_waiters_queue_depth: Lazy<RwLock<HashMap<usize, AtomicU64>>>,
```

Comment:
Is having a lock here too heavy?

Reply:
Hard to tell. Sytems are re-used, so we shouldn't do sys-calls generally when acquiring the read lock for updates. However, it would be quite hard to diagnose contention on this lock.

Reply:
we now use a new design